Lib/_pyrepl/utils.py

```diff
@@ -73,5 +174,19 @@ def disp_str(buffer: str) -> tuple[CharBuffer, CharWidths]:
         else:
             chars.append(c)
             char_widths.append(str_width(c))
+
+        if colors and colors[0].span.end == i:  # current color ends now
+            post_color = TAG_TO_ANSI["SYNC"]
+            colors.pop(0)
+
+        chars[-1] = pre_color + chars[-1] + post_color
```

Comment:
Not sure if this matters but this being in a tight loop may be a performance problem